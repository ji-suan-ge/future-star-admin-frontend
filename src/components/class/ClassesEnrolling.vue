<template>
  <div>
    <h4 v-if="classes.length === 0">暂无招生中班级</h4>
    <el-col :span="12" v-for="(item) in classes" :key="item.id">
      <el-card>
        <el-row slot="header" type="flex" align="middle">
          <el-col :span="14">
            {{ item.className }}
          </el-col>
          <el-col :span="2">
            <el-button type="success" size="small" icon="el-icon-check" circle></el-button>
          </el-col>
          <el-col :span="2">
            <el-button type="primary" size="small" icon="el-icon-more" @click="lookOverClassDetail" circle></el-button>
          </el-col>
          <el-col :span="2">
            <el-button type="primary" size="small" icon="el-icon-edit-outline" @click="editClassInfo" circle></el-button>
          </el-col>
          <el-col :span="2">
            <el-button type="primary" size="small" icon="el-icon-s-custom" @click="enrollmentAudit" circle></el-button>
          </el-col>
          <el-col :span="2">
            <el-button type="danger" size="small" icon="el-icon-delete" circle></el-button>
          </el-col>
        </el-row>
        <el-row>
          起止日期：{{ item.classStartTime }}—— {{ item.classEndTime }}
        </el-row>
        <el-row>
          招生人数：{{ item.currentAmountOfStudent }} / {{ item.capacity }}
        </el-row>
        <el-row>
          班级状态：{{ item.state }}
        </el-row>
        <el-row>
          班级简介：{{ item.classIntroduction | ellipsis }}
        </el-row>
      </el-card>
    </el-col>
  </div>
</template>
<script>
export default {
  name: 'EnrollingClasses',
  data () {
    return {
      classes: [
        {
          id: '2',
          className: '素质教育专题班',
          classIntroduction: '这里是一个加了长文本省略号替代的班级介绍',
          classStartTime: '2019/12/20',
          classEndTime: '2019/12/21',
          currentAmountOfStudent: '0',
          capacity: '15',
          state: '招生中'
        },
        {
          id: '1',
          className: '素质教育专题班',
          classIntroduction: '这里是一个加了长文本省略号替代的班级介绍',
          classStartTime: '2019/12/20',
          classEndTime: '2019/12/21',
          currentAmountOfStudent: '0',
          capacity: '15',
          state: '招生中'
        },
        {
          id: '3',
          className: '素质教育专题班',
          classIntroduction: '这里是一个加了长文本省略号替代的班级介绍',
          classStartTime: '2019/12/20',
          classEndTime: '2019/12/21',
          currentAmountOfStudent: '0',
          capacity: '15',
          state: '招生中'
        }
      ]
    }
  },
  methods: {
    editClassInfo: function () {
      this.$router.push('/edit-class-info')
    },
    lookOverClassDetail: function () {
      this.$router.push('/class-detail')
    },
    enrollmentAudit: function () {
      this.$router.push('/enrollment-audit')
    }
  },
  filters: {
    ellipsis (text) {
      if (!text) return ''
      if (text.length > 30) {
        return text.slice(0, 30) + '......'
      }
      return text
    }
  }
}
</script>

<style scoped>
  div {
    padding: 0;
    margin: 0;
  }

  .el-card {
    margin: 10px;
  }

  .el-button {
    float: right;
  }
</style>
