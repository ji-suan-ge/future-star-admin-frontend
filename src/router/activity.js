import CreateActivityPage from '../page/CreateActivityPage'
import ShowActivityParticipantsPage from '../page/ShowActivityParticipantsPage'
import EditExistingActivityPage from '../page/EditExistingActivityPage'
import ShowActivityInformationPage from '../page/ShowActivityInformationPage'

export default [
  {
    path: '/create-activity',
    name: 'CreateActivityPage',
    component: CreateActivityPage
  },
  {
    path: '/show-participants',
    name: 'ShowActivityParticipantsPage',
    component: ShowActivityParticipantsPage
  },
  {
    path: '/edit-activity',
    name: 'EditExistingActivityPage',
    component: EditExistingActivityPage
  },
  {
    path: '/show-activity-details',
    name: 'ShowActivityInformationPage',
    component: ShowActivityInformationPage
  }
]
