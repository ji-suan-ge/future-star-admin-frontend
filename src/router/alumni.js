import ShowAlumniInformationPage from '../page/alumni-management/ShowAlumniInformationPage'
import EditAlumniInformationPage from '../page/alumni-management/EditAlumniInformationPage'
import ShowAlumniEditHistoryPage from '../page/alumni-management/ShowAlumniEditHistoryPage'
import ShowAlumniHistoryInformationPage from '../page/alumni-management/ShowAlumniHistoryInformationPage'

export default [
  {
    path: '/show-all-information',
    name: 'ShowAlumniInformationPage',
    component: ShowAlumniInformationPage
  },
  {
    path: '/edit-alumni-information',
    name: 'EditAlumniInformationPage',
    component: EditAlumniInformationPage
  },
  {
    path: '/show-edit-history',
    name: 'ShowAlumniEditHistoryPage',
    component: ShowAlumniEditHistoryPage
  },
  {
    path: '/show-history-information',
    name: 'ShowAlumniHistoryInformationPage',
    component: ShowAlumniHistoryInformationPage
  }
]
