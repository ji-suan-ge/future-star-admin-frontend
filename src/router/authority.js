import AddAdministratorPage from '../page/authority-management/AddAdministratorPage'

export default [
  {
    path: '/add-administrator',
    name: 'AddAdministratorPage',
    component: AddAdministratorPage
  }
]
