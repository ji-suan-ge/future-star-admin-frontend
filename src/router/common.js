import LoginPage from '../page/LoginPage'
import HomePage from '../page/HomePage'
import homeChildren from './home-children'

export default [
  {
    path: '/',
    name: 'Root',
    redirect: '/login'
  },
  {
    path: '/login',
    name: 'LoginPage',
    component: LoginPage
  },
  {
    path: '/home',
    name: 'HomePage',
    component: HomePage,
    redirect: '/home/welcome',
    children: [
      ...homeChildren
    ]
  }
]
