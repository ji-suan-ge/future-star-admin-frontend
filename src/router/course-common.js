import DetailOfSemesterPage from '../page/course-common/DetailOfSemesterPage'
import DetailOfClassPage from '../page/course-common/DetailOfClassPage'
import DetailOfCoursePage from '../page/course-common/DetailOfCoursePage'
import DetailOfTeacherPage from '../page/course-common/DetailOfTeacherPage'
import DetailOfSubCoursePage from '../page/course-common/DetailOfSubCoursePage'
import EditSemesterInfoPage from '../page/course-common/EditSemesterInfoPage'
import EditClassInfoPage from '../page/course-common/EditClassInfoPage'
import EditCourseInfoPage from '../page/course-common/EditCourseInfoPage'
import EditTeacherInfoPage from '../page/course-common/EditTeacherInfoPage'
import AddCoursePage from '../page/course-common/AddCoursePage'
import AddCourseResourcePage from '../page/course-common/AddCourseResourcePage'
import StudentInClass from '../page/course-common/StudentInClass'

export default [
  {
    path: '/semester-detail',
    name: 'DetailOfSemesterPage',
    component: DetailOfSemesterPage
  },
  {
    path: '/class-detail',
    name: 'DetailOfClassPage',
    component: DetailOfClassPage
  },
  {
    path: '/course-detail',
    name: 'DetailOfCoursePage',
    component: DetailOfCoursePage
  },
  {
    path: '/teacher-detail',
    name: 'DetailOfTeacherPage',
    component: DetailOfTeacherPage
  },
  {
    path: '/sub-course-detail',
    name: 'DetailOfSubCoursePage',
    component: DetailOfSubCoursePage
  },
  {
    path: '/edit-semester-info',
    name: 'EditSemesterInfoPage',
    component: EditSemesterInfoPage
  },
  {
    path: '/edit-class-info',
    name: 'EditClassInfoPage',
    component: EditClassInfoPage
  },
  {
    path: '/edit-course-info',
    name: 'EditCourseInfoPage',
    component: EditCourseInfoPage
  },
  {
    path: '/edit-teacher-info',
    name: 'EditTeacherInfoPage',
    component: EditTeacherInfoPage
  },
  {
    path: '/add-course',
    name: 'AddCoursePage',
    component: AddCoursePage
  },
  {
    path: '/add-course-resource',
    name: 'AddCourseResourcePage',
    component: AddCourseResourcePage
  },
  {
    path: '/student-in-class',
    name: 'StudentInClass',
    component: StudentInClass
  }
]
