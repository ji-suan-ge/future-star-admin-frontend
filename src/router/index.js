import Vue from 'vue'
import VueRouter from 'vue-router'
import LoginPage from '../page/LoginPage'
import HomePage from '../page/HomePage'
import EnrollmentManagementPage from '../page/EnrollmentManagementPage'
import CourseManagementPage from '../page/CourseManagementPage'
import ActivityManagementPage from '../page/ActivityManagementPage'
import AlumniManagementPage from '../page/AlumniManagementPage'
import AuthorityManagementPage from '../page/AuthorityManagementPage'
import CreateActivityPage from '../page/CreateActivityPage'
import AddAdministrator from '../page/AddAdministrator'
import WelcomePage from '../page/WelcomPage'

Vue.use(VueRouter)

const routes = [
  {
    path: '/',
    name: 'Root',
    redirect: '/login'
  },
  {
    path: '/login',
    name: 'LoginPage',
    component: LoginPage
  },
  {
    path: '/home',
    name: 'HomePage',
    component: HomePage,
    redirect: '/home/welcome',
    children: [
      {
        path: 'welcome',
        name: 'WelcomePage',
        component: WelcomePage
      },
      {
        path: 'enrollment-management',
        name: 'EnrollmentManagementPage',
        component: EnrollmentManagementPage
      },
      {
        path: 'course-management',
        name: 'CourseManagementPage',
        component: CourseManagementPage
      },
      {
        path: 'activity-management',
        name: 'ActivityManagementPage',
        component: ActivityManagementPage
      },
      {
        path: 'alumni-management',
        name: 'AlumniManagementPage',
        component: AlumniManagementPage
      },
      {
        path: 'authority-management',
        name: 'AuthorityManagementPage',
        component: AuthorityManagementPage
      },
      {
        path: 'add-administrator',
        name: 'AddAdministrator',
        component: AddAdministrator
      }
    ]
  },
  {
    path: '/create-activity',
    name: 'CreateActivityPage',
    component: CreateActivityPage
  }
]

const router = new VueRouter({
  routes
})

export default router
