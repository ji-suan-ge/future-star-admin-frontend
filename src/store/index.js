import Vue from 'vue'
import Vuex from 'vuex'

Vue.use(Vuex)

export default new Vuex.Store({
  state: {
    activeNameOfCourse: 'first',
    activeNameOfEnrollment: 'first'
  },
  mutations: {
    changeActiveNameOfEnrollment (state, activeName) {
      state.activeNameOfEnrollment = activeName
    },
    changeActiveNameOfCourse (state, activeName) {
      state.activeNameOfCourse = activeName
    }
  },
  actions: {
    changeActiveNameOfEnrollment (context, activeName) {
      context.commit('changeActiveNameOfEnrollment', activeName)
    },
    changeActiveNameOfCourse (context, activeName) {
      context.commit('changeActiveNameOfCourse', activeName)
    }
  },
  getters: {
    getActiveNameOfEnrollment (state) {
      return state.activeNameOfEnrollment
    },
    getActiveNameOfCourse (state) {
      return state.activeNameOfCourse
    }
  },
  modules: {
  }
})
