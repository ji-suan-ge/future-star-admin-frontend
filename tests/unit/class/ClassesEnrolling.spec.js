import { createLocalVue, mount } from '@vue/test-utils'
import ClassesEnrolling from '../../../src/components/class/ClassesEnrolling'
import Vuex from 'vuex'
import ElementUI from 'element-ui'
import VueRouter from 'vue-router'
import store from '../../../src/store'
import router from '../../../src/router'

describe('ClassesEnrolling', () => {
  const localVue = createLocalVue()
  localVue.use(Vuex)
  localVue.use(ElementUI)
  localVue.use(VueRouter)
  const wrapper = mount(ClassesEnrolling, {
    store,
    router,
    localVue
  })
  test('快照测试', async () => {
    await expect(wrapper.html()).toMatchSnapshot()
  })
  test('是一个 Vue 实例', async () => {
    await expect(wrapper.isVueInstance()).toBeTruthy()
  })
})
