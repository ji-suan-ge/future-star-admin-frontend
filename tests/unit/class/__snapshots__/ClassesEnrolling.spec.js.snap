// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClassesEnrolling 快照测试 1`] = `
<div>
  <!---->
  <div class="el-col el-col-12">
    <div class="el-card is-always-shadow">
      <div class="el-card__header">
        <div class="el-row is-align-middle el-row--flex">
          <div class="el-col el-col-14">
            素质教育专题班
          </div>
          <div class="el-col el-col-2"><button type="button" class="el-button el-button--primary el-button--small is-circle">
              <!----><i class="el-icon-check"></i>
              <!----></button></div>
          <div class="el-col el-col-2"><button type="button" class="el-button el-button--primary el-button--small is-circle">
              <!----><i class="el-icon-more"></i>
              <!----></button></div>
          <div class="el-col el-col-2"><button type="button" class="el-button el-button--primary el-button--small is-circle">
              <!----><i class="el-icon-edit-outline"></i>
              <!----></button></div>
          <div class="el-col el-col-2"><button type="button" class="el-button el-button--primary el-button--small is-circle">
              <!----><i class="el-icon-s-custom"></i>
              <!----></button></div>
          <div class="el-col el-col-2"><button type="button" class="el-button el-button--danger el-button--small is-circle">
              <!----><i class="el-icon-delete"></i>
              <!----></button></div>
        </div>
      </div>
      <div class="el-card__body">
        <div class="el-row">
          起止日期：2019/12/20—— 2019/12/21
        </div>
        <div class="el-row">
          招生人数：0 / 15
        </div>
        <div class="el-row">
          班级状态：招生中
        </div>
        <div class="el-row">
          班级简介：这里是一个加了长文本省略号替代的班级介绍
        </div>
      </div>
    </div>
  </div>
  <div class="el-col el-col-12">
    <div class="el-card is-always-shadow">
      <div class="el-card__header">
        <div class="el-row is-align-middle el-row--flex">
          <div class="el-col el-col-14">
            素质教育专题班
          </div>
          <div class="el-col el-col-2"><button type="button" class="el-button el-button--primary el-button--small is-circle">
              <!----><i class="el-icon-check"></i>
              <!----></button></div>
          <div class="el-col el-col-2"><button type="button" class="el-button el-button--primary el-button--small is-circle">
              <!----><i class="el-icon-more"></i>
              <!----></button></div>
          <div class="el-col el-col-2"><button type="button" class="el-button el-button--primary el-button--small is-circle">
              <!----><i class="el-icon-edit-outline"></i>
              <!----></button></div>
          <div class="el-col el-col-2"><button type="button" class="el-button el-button--primary el-button--small is-circle">
              <!----><i class="el-icon-s-custom"></i>
              <!----></button></div>
          <div class="el-col el-col-2"><button type="button" class="el-button el-button--danger el-button--small is-circle">
              <!----><i class="el-icon-delete"></i>
              <!----></button></div>
        </div>
      </div>
      <div class="el-card__body">
        <div class="el-row">
          起止日期：2019/12/20—— 2019/12/21
        </div>
        <div class="el-row">
          招生人数：0 / 15
        </div>
        <div class="el-row">
          班级状态：招生中
        </div>
        <div class="el-row">
          班级简介：这里是一个加了长文本省略号替代的班级介绍
        </div>
      </div>
    </div>
  </div>
  <div class="el-col el-col-12">
    <div class="el-card is-always-shadow">
      <div class="el-card__header">
        <div class="el-row is-align-middle el-row--flex">
          <div class="el-col el-col-14">
            素质教育专题班
          </div>
          <div class="el-col el-col-2"><button type="button" class="el-button el-button--primary el-button--small is-circle">
              <!----><i class="el-icon-check"></i>
              <!----></button></div>
          <div class="el-col el-col-2"><button type="button" class="el-button el-button--primary el-button--small is-circle">
              <!----><i class="el-icon-more"></i>
              <!----></button></div>
          <div class="el-col el-col-2"><button type="button" class="el-button el-button--primary el-button--small is-circle">
              <!----><i class="el-icon-edit-outline"></i>
              <!----></button></div>
          <div class="el-col el-col-2"><button type="button" class="el-button el-button--primary el-button--small is-circle">
              <!----><i class="el-icon-s-custom"></i>
              <!----></button></div>
          <div class="el-col el-col-2"><button type="button" class="el-button el-button--danger el-button--small is-circle">
              <!----><i class="el-icon-delete"></i>
              <!----></button></div>
        </div>
      </div>
      <div class="el-card__body">
        <div class="el-row">
          起止日期：2019/12/20—— 2019/12/21
        </div>
        <div class="el-row">
          招生人数：0 / 15
        </div>
        <div class="el-row">
          班级状态：招生中
        </div>
        <div class="el-row">
          班级简介：这里是一个加了长文本省略号替代的班级介绍
        </div>
      </div>
    </div>
  </div>
</div>
`;
